tools/sphere.c

```diff
@@ -80,3 +78,31 @@ int do_touch( string str ) {
 int do_gaze( string str ) {
 
 }
+
+/**
+ *  Function finds and returns the sloop shadow based on the
+ *  string passed.
+ *  @param {String} sloop_type : valid values either "slooper" or "slooped"
+ *  @param {Object} ob :  player object to test for shadows
+ *  @return {Object} : returns the shadow object
+ */
+
+string find_sloop_shadow( string sloop_type, object ob ) {
+    if ( sloop_type != "slooper" || sloop_type != "slooped" ) {
```

Comment:
This is another example of "magic values", which if you remember, should be declared as #defines. 

But actually, if this were me, I'd split this up into two functions: find_slooper_shadow() and find_sloopee_shadow(). The problem with using magic values in your public API is that it requires anyone who calls your function to also know what those magic values are (usually by #include the ".h" file where your #define statements are). It's error prone and introduces ambiguity into your API which is usually something you want to avoid. It also isn't any more concise for the caller, they still have to type the same number of chars.

Now, you might be thinking it's less concise to me, the person coding the API, to have two different functions which do pretty much the same thing. Your instinct to combine the code for both into a single function is a good one. That's called code re-use, and we can still do it, even if we have two different public functions. 

To do this, we move the common logic into a third, private function, which the public functions can then call under-the-hood. Something like this:

```
string find_slooper_shadow(object ob) {
  return find_sloop_shadow(ob, SLOOPER_SHAD);
}

string find_sloopee_shadow(object ob) {
  return find_sloop_shadow(ob, SLOOPEE_SHAD);
}

string find_sloop_shadow(object ob, string sloop_shadow_path) {
  your loop goes here...
}
```

I did a few things here. First, I split your one public function up like I talked about earlier. But also, instead of introducing two new magic values for "slooper" and "slooped", I used the magic values we already had for the shadow paths. I'll let you figure out how the loop would have to change to accommodate that.